QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question){
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public bool TryUpdateRecord(int id, string answer){
+            IRecord recordToUpdate;
+            
+            try{
+                recordToUpdate = allRecords.First(recordItem => recordItem.ID == id);
```

Comment:
If I were to go back to FirstOrDefault, I could do something like:

```
recordToUpdate = allRecords.FirstOrDefault(recordItem => recordItem.ID == id);
if(recordToUpdate == default(Record){
     return false;
}

```

Using just .First would give me an error otherwise

I've also tried .`Where`, `.Contains`, `.Exists`, but those want an IRecord as a parameter rather than my lambda expression.